### PR TITLE
Update OIDC and JWT dependencies

### PR DESCRIPTION
**Pull request by n1v0lg**
Updates internal dependencies for OIDC and JWT realms.

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticator.java [hunk 1]
```diff
@@ -135,6 +138,15 @@
  */
 public class OpenIdConnectAuthenticator {
 
+    static {
+        JSONValue.registerWriter(Nonce.class, new JsonWriterI<Nonce>() {
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticator.java:142]: json-smart uses reflection for serializing classes, unless they have a registered writer. This is turn would require createClassLoader permissions. 

`Nonce` is the only class where this is potentially an case (and the issue only surfaces in unit test code where we manually construct claim sets involving nonces as claims). Registering a write here lets us avoid the policy grants.

**Comment by n1v0lg** [comment]
@elasticmachine update branch

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticator.java (hunk 1 above)
Comment (n1v0lg) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticator.java:142]: This needs a better spot though...

**Review by jakelandis — APPROVED** [review]
LGTM

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Pull request merged**